### DISCUSSION TITLE
Can i have two primary keys on the same table?

### DISCUSSION THREAD

**Discussion by rafaelrossa27** [ujjwalguptaofficial/JsStore#350] [Q&A]
Can i have two primary keys on the same table?
ex.
var tbRepcli = {
    name: 'Repcli',
    columns: {
        rep_codigo: { primaryKey: true, dataType: "number"},
        cli_codigo: { primaryKey: true, dataType: "number"},
        rep_ativo: { notNull: true, dataType: "string" }
    }
};

I'm trying to insert these rows:

[4, 56, "SIM"];
[4, 87, "SIM"];

and this error ocurred:
{message: "Unable to add key to index 'rep_codigo': at least …key does not satisfy the uniqueness requirements.", type: 'ConstraintError'}

**Comment by ujjwalguptaofficial** [comment]
There is always one primary key in one table. 

Why do you want two primary key ?

**Reply by rafaelrossa27** [reply]
This is an n:n table, I have a customer table and an agent table, this table connects the two, both the customer code and the agent code will appear more than once, but you can never duplicate the same customer code with the same agent code.

thanks for reply

**Reply by ujjwalguptaofficial** [reply]
i see, try keyPath with unique. 

keypath allows you to add multiple indexing and unique allows you to have unique data. 

doc - https://jsstore.net/docs/database/column

**Reply by rafaelrossa27** [reply]
OK, i will try these options.

Thanks!
